### PR TITLE
Add namespaces in knative/pkg

### PR DESCRIPTION
**Pull request by zxDiscovery**
synchronize with https://github.com/knative/serving/pull/2708

### CODE REVIEW

File: system/names.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - name: %s
+    valueFrom:
+      fieldRef:
+        fieldPath: metadata.namespace
+
+If this is a Go unit test consuming system.Namespace() then it should add the
+following import:
+
+import (
+	_ "github.com/knative/serving/pkg/system/testing"
```
Comment (mattmoor) [inline system/names.go]: drop serving.

File: system/names.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline system/names.go]: 2019 here and elsewhere

File: system/testing/names.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"os"
+
+	"github.com/knative/serving/pkg/system"
+)
+
+func init() {
+	os.Setenv(system.NamespaceEnvKey, "knative-testing")
+
```
Comment (mattmoor) [inline system/testing/names.go]: nit: delete empty line

Comment (zxDiscovery) [reply system/testing/names.go]: Done

File: system/names.go (hunk 2 above)
Comment (zxDiscovery) [reply system/names.go]: Done

File: system/names.go (hunk 1 above)
Comment (zxDiscovery) [reply system/names.go]: Done

File: system/names.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+package system
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	NamespaceEnvKey = "SYSTEM_NAMESPACE"
```
[the remark this answers is not in the document]
Comment (zxDiscovery) [reply system/names.go]: Add prefixing sounds good. If we change to `KN_SYSTEM_NAMESPACE` here, do we need to synchronize this to the config yaml file in `knative/serving/config`, `knative/build/config` and `knative/eventing/config`.

[the remark this answers is not in the document]
Comment (mattmoor) [reply system/names.go]: If anything I'd go with `KNATIVE_`, but I could see myself using this package beyond Knative controllers too, so having the key be agnostic could also be useful.

**Comment by mattmoor** [comment]
/lgtm
/approve
/hold

Use `/hold cancel` to unblock submission.  If you want to change the environment variable, then I'll need to give a fresh `/lgtm`.

**Pull request merged**